### PR TITLE
Add `attach last upper to the next token` implementation

### PR DESCRIPTION
**Pull request by irinakhismatullina**
Here's the reason for the feature introduction #403.

### CODE REVIEW

File: sourced/ml/algorithms/token_parser.py [hunk 1]
```diff
@@ -33,18 +33,22 @@ class TokenParser:
     # if True we have only ["sourc", "algorithm"].
     # if you do not want to filter small tokens set min_split_length=1.
     SAVE_TOKEN_STYLE = False  # whether yield metadata that can be used to reconstruct initial
-    # identifier
+    # identifier.
+    ATTACH_UPPER = False  # True to attach the last of several uppercase letters in a row to
```
Comment (vmarkovtsev) [inline sourced/ml/algorithms/token_parser.py]: `True` should be the default.

File: sourced/ml/algorithms/token_parser.py [hunk 2]
```diff
@@ -33,18 +33,22 @@ class TokenParser:
...
+    # the next token. Example: 'HTMLResponce' -> ["html", "responce"] if True,
```
Comment (vmarkovtsev) [inline sourced/ml/algorithms/token_parser.py]: respon**s**e

File: sourced/ml/tests/test_token_parser.py [hunk 3]
```diff
@@ -53,6 +53,49 @@ def test_process_token(self):
...
+            ("SmallIdsToFoOo", ["small", "ids", "to", "fo", "oo"]),
+            ("SmallIdFooo", ["small", "id", "fooo"]),
+            ("ONE_M0re_.__badId.example", ["one", "m", "re", "bad",
+                                           "id", "example"]),
+            ("never_use_Such__varsableNames", ["never", "use", "such", "varsable", "names"]),
+            ("a.b.c.d", ["a", "b", "c", "d"]),
+            ("A.b.Cd.E", ["a", "b", "cd", "e"]),
+            ("looong_sh_loooong_sh", ["looong", "sh", "loooong", "sh"]),
+            ("sh_sh_sh_sh", ["sh", "sh", "sh", "sh"]),
+            ("loooong_loooong_loooong", ["loooong", "loooong", "loooong"])
```
Comment (zurk) [inline sourced/ml/tests/test_token_parser.py]: Let's add `ONLYCAPS` and `nocaps` tokens to both tests to see that they remain the same.

**Comment by zurk** [comment]
looks like `ONLYCAPS` does not work.

**Comment by irinakhismatullina** [comment]
There's stemming in some tests, fixed now.

**Comment by irinakhismatullina** [comment]
@zurk ready

**Comment by zurk** [comment]
Vadim's review was fulfilled so I am merging.

**Pull request merged**